app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx

```diff
@@ -0,0 +1,278 @@
+
+    return () =>
+      document.removeEventListener('click', closeOnClickOutsideListener);
+  }, [searchTerm]);
+
+  useLayoutEffect(() => {
+    inputRef.current.focus();
+    inputRef.current.setSelectionRange(cursorPosition, cursorPosition - 1);
+  }, [cursorPosition]);
+
+  const handleValueChange = ({ target: { value } }) => {
+    setTextContent(value);
+    const { isUserMention, indexOfMentionStart } = getMentionWordData(
+      inputRef.current,
+    );
+
+    const { selectionStart } = inputRef.current;
+
+    if (isUserMention) {
+      // search term begins after the @ character
+      const searchTermStartPosition = indexOfMentionStart + 1;
+
+      const mentionText = value.substring(
+        searchTermStartPosition,
+        selectionStart,
+      );
+
+      const { x: cursorX, y } = getCursorXY(
+        inputRef.current,
+        indexOfMentionStart,
+      );
+      const textAreaX = inputRef.current.offsetLeft;
+
+      // On small screens always show dropdown at start of textarea
+      const dropdownX = isSmallScreen ? textAreaX : cursorX;
+
+      setDropdownPositionPoints({ x: dropdownX, y });
+      setSearchTerm(mentionText);
+      setSelectionInsertIndex(searchTermStartPosition);
+    } else if (searchTerm) {
+      // User has moved away from an in-progress @mention - clear current search
+      setSearchTerm('');
+      setAriaHelperText('');
+      setUsers([]);
+    }
+  };
+
+  const handleSelect = (username) => {
+    const textWithSelection = `${textContent.substring(
+      0,
+      selectionInsertIndex,
+    )}${username}${textContent.substring(inputRef.current.selectionStart)}`;
+
+    // Clear the current search
+    setSearchTerm('');
+    setUsers([]);
+    setAriaHelperText('');
+
+    // Update the text area value
+    setTextContent(textWithSelection);
+
+    // Update the cursor to directly after the selection
+    const newCursorPosition = selectionInsertIndex + username.length + 1;
+    setCursorPosition(newCursorPosition);
+  };
+
+  useLayoutEffect(() => {
+    if (inputRef.current) {
+      // Replace the whole textarea passed in props with the autocomplete textarea
+      replaceTextArea(replaceElement, inputRef.current);
+    }
+  }, [replaceElement]);
+
+  return (
+    <Fragment>
+      <div aria-live="polite" class="screen-reader-only">
+        {ariaHelperText}
+      </div>
+      <Combobox
+        id="combobox-container"
+        onSelect={handleSelect}
+        className="crayons-autocomplete"
+      >
+        <ComboboxInput
+          ref={inputRef}
+          value={textContent}
+          data-mention-autocomplete-active="true"
+          as="textarea"
+          autocomplete={false}
+          onChange={handleValueChange}
+        />
+        {searchTerm && (
+          <ComboboxPopover
+            className="crayons-autocomplete__popover"
+            id="mention-autocomplete-popover"
+            style={{
+              position: 'absolute',
```

Comment:
Refactor (non-blocking): Instead of setting the position with inline styles, use the [absolute CSS utility class](https://storybook.forem.com/?path=/story/utility-first-classes-position--absolute).